HW1.py

```diff
@@ -15,4 +15,118 @@
 4) Сгенерировать dict() из списка ключей ниже по формуле (key : key* key).
 keys = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
 ожидаемый результат: {1: 1, 2: 4, 3: 9 …}
-"""
\ No newline at end of file
+"""
+
+# 1.
+InitialArray = [10, 11, 2, 3, 5, 8, 23, 11, 2, 5, 76, 43, 2, 32, 76, 3, 10, 0, 1]
+# 1.1) убрать из него повторяющиеся элементы
+NoDuplicateArray = list(set(InitialArray))
+print("1.1 No duplicates array")
+print(NoDuplicateArray)
+
+# The function to check that we were right
+def FindDuplicateInArray(TheArray):
+    while bool(TheArray) is True:
+        ComparisonTarget = TheArray.pop()
+        # print (ComparisonTarget)
+        # print(TheArray)
+        for element in TheArray:
+            # print(f"comparing {ComparisonTarget} with {element}")
+            if ComparisonTarget == element:
+                print(f"{ComparisonTarget} is the duplicate!")
+                return
+            # print("--------------")
+    print("No duplicates were found")
+# FindDuplicateInArray(NoDuplicateArray)
+
+# 1.2) вывести 3 наибольших числа из исходного массива
+def NMaxElents(HowManyElements, TheArray):
+    MaxNumbersList = []
+    NoDuplicateArray = list(set(TheArray))  # to avoid the "76, 76, 43" answer
+    while bool(HowManyElements) is True:
+        HowManyElements -= 1
+        MaxNumbersList.append(max(NoDuplicateArray))
+        NoDuplicateArray.remove(max(NoDuplicateArray))
+    return(print(MaxNumbersList))
+print("1.2 Three max values of the array")
+NMaxElents(3, InitialArray)
```

Comment:
could be shorted 
```
sorted_list = sorted(the_array, reversed=True)
print(sorted_list[:3])
```

Reply:
There is a typo: should be without 'd': reverse=True